src/lib.rs

```diff
@@ -975,3 +1039,94 @@ pub async fn validate_and_load_static_assets<'a>(
         Err(e) => Err(e),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use goose::config::GooseConfiguration;
+    use goose::goose::get_base_url;
+    use gumdrop::Options;
+
+    const EMPTY_ARGS: Vec<&str> = vec![];
+    const HOST: &str = "http://example.com";
+
+    #[tokio::test]
+    async fn get_static_elements() {
+        const HTML: &str = r#"<!DOCTYPE html>
+        <html>
+        <body>
+            <!-- 3 valid CSS paths -->
```

Comment:
Are there additional valid or invalid test cases we should be testing here?

Reply:
Maybe let's add a query argument to a css/js path.
```
<link href="http://example.com/example.css?version=abc123" rel="stylesheet" />
```

Reply:
Excellent suggestion, I added a couple of different examples with query parameters. 👍